lib/Logger/Logger.cpp

```diff
@@ -663,6 +666,25 @@ void Logger::log(char const* logid, char const* function, char const* file,
 
     TRI_ASSERT(offset == 0);
   } else {
+    std::unique_ptr<GeneralEscaper> escaper;
+    if (_useControlEscaped) {
+      if (_useUnicodeEscaped) {
+        escaper = std::make_unique<
+            Escaper<ControlCharsEscaper, UnicodeCharsEscaper>>();
+      } else {
+        escaper = std::make_unique<
+            Escaper<ControlCharsEscaper, UnicodeCharsRetainer>>();
+      }
+    } else {
+      if (_useUnicodeEscaped) {
+        escaper = std::make_unique<
+            Escaper<ControlCharsSuppressor, UnicodeCharsEscaper>>();
+      } else {
+        escaper = std::make_unique<
+            Escaper<ControlCharsSuppressor, UnicodeCharsRetainer>>();
+      }
+    }
```

Comment:
It looks like we are now creating one Escaper instance on the heap for each log message. We shouldn't do that.
The logging is a critical part of code that needs to be a fast as possible.